src/unix/fs.c

```diff
@@ -860,7 +860,11 @@ static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_processor process)
 
   if (bufs != req->bufsml)
     uv__free(bufs);
-  req->bufs = NULL;
+
+  if (!(errno == EINTR && total == -1)) {
```

Comment:
I believe that if this condition will be placed this way - this function will free the `bufs` variable, causing use-after-free on next retry. What you probably want to do is to negate the `test` expression, and just make it return `total`, and put all of this before line `861`.